src/tango/xdp/fd_xsk.c

```diff
@@ -730,12 +789,7 @@ fd_xsk_tx_enqueue( fd_xsk_t *            xsk,
 
   /* sz is min( available, count ) */
   uint sz = cap - ( prod - cons );
-  /* TODO this doesn't work as expected
-     if we early exit here, no wakeup occurs, sendto doesn't get called again
-     and the ring doesn't get serviced
-     This implies we need to call sendto AGAIN even if the ring hasn't changed
   if( sz == 0 )    return 0;
```

Comment:
This needs to be removed. If a prior call did not have "flush" set, the ring may be full and yet the driver may not know it. Calling with "flush" should invoke wakeup regardless.